src/libcollections/fmt.rs

```diff
@@ -407,6 +407,21 @@
 //! For floating-point types, this indicates how many digits after the decimal
 //! point should be printed.
 //!
+//! By example, these:
+//!
+//! ```
+//! println!("Hello {1} is {2:.0$}", 5, "x", 0.01);
+//! println!("Hello {0} is {2:.1$}", "x", 5, 0.01);
+//! println!("Hello {} is {:.*}",    "x", 5, 0.01);
+//! println!("Hello {} is {2:.*}",   "x", 5, 0.01);
```

Comment:
Perhaps have a comment above each one reading out what each one means? Something like "Hello {first argument (x)} is {second argument (0.01) with precision specified in zeroeth argument (5)}"